segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java

```diff
@@ -90,6 +92,10 @@
 
     private FileModificationMonitor tlsCertFileModificationMonitor; // used only if tls reload is enabled
 
+    // Used for running token expiry handling tasks.
+    private final ScheduledExecutorService tokenExpiryHandlerExecutor =
```

Comment:
Please do not create random executors. There are already executors created in the Segment Store process; use one of them. Or otherwise try to rework your code to not require an executor.

Reply:
Ok. I have modified the code to use a preexisting executor that is passed from the `ServiceStarter`.

Reply:
On second thought, I realized that it'd be better to run the token expiry tasks on a separate executor spinning up lower priority threads, just so that these tasks don't end up overwhelming the core executor. So, I have now modified the code again. 

Please take a look again and see what you think.